ui/pages/confirm-transaction-base/confirm-transaction-base.container.js

```diff
@@ -84,10 +89,13 @@ const mapStateToProps = (state, ownProps) => {
     nextNonce,
     provider: { chainId },
   } = metamask;
+
+  const transactionsToDisplayOnFailure = getFailedTransactionsToDisplay(state);
+
   const { tokenData, txData, tokenProps, nonce } = confirmTransaction;
   const { txParams = {}, id: transactionId, type } = txData;
   const transaction =
-    Object.values(unapprovedTxs).find(
+    Object.values({ ...unapprovedTxs, ...transactionsToDisplayOnFailure }).find(
```

Comment:
In this case we want to ensure that the unapproved transaction is always chosen first, if it's present in both collections. It probably works correctly here because `Object.values` returns entries in insert order in most cases. But I do not think the iteration order of `Object.values` is guaranteed to be insert-order in all cases. I don't know if this works consistently in all of our supported browsers.

To make certain that this check is done in the correct order, we should look in each of these collections separately:

e.g. something like:

```
const transaction =
  Object.values(unapprovedTxs).find(({ id }) => id === (transactionId || Number(paramsTransactionId))) ||
  Object.values(transactionsToDisplayOnFailure).find(({ id }) => id === (transactionId || Number(paramsTransactionId))) ||
  {};
```